src/app-layer-dns-common.c

```diff
@@ -253,11 +253,11 @@ int DNSGetTxLogged(void *alstate, void *tx, uint32_t logger)
 
 /** \brief get value for 'complete' status in DNS
  *
- *  For DNS we use a simple bool. 1 means done.
+ *  For DNS 1 means request done, 2 means reply done (or lost).
  */
 int DNSGetAlstateProgressCompletionStatus(uint8_t direction)
 {
-    return 1;
+    return 2;
```

Comment:
No, because with WithProgress we want it to fire in specific states, one where that state is 0.  If we use 0 as a progress state when registering, it does not get accepted as 0 due to this: 

https://github.com/inliniac/suricata/blob/master/src/output-tx.c#L88

So while want to get called when tc is 0, and ts is 1, the above will force it to be 1 and 1. So implementing tri-state for the specific protocol seemed easier then changing the tx handling logic.